politeiad/cache/cockroachdb/decred.go

```diff
@@ -884,6 +884,243 @@ func (d *decred) cmdTokenInventory(payload string) (string, error) {
+// begun the voting process. Results are lazily loaded into this table, so
+// some results will need to be manually looked up in the CastVote table.
+func (d *decred) getVoteResultsForStartVotes(svMap map[string]StartVote) (map[string][]decredplugin.VoteOptionResult, error) {
+
+	resMap := make(map[string][]decredplugin.VoteOptionResult)
+
+	if len(svMap) == 0 {
+		return resMap, nil
+	}
+
+	tokens := make([]string, 0, len(svMap))
+	for token := range svMap {
+		tokens = append(tokens, token)
+	}
+
+	vrs := make([]VoteResults, 0, len(tokens))
+	err := d.recordsdb.
+		Where("token IN (?)", tokens).
+		Preload("Results").
+		Preload("Results.Option").
+		Find(&vrs).
+		Error
+	if err != nil {
+		return nil, err
+	}
+
+	for _, vr := range vrs {
+		resMap[vr.Token] = convertVoteOptionResultsToDecred(vr.Results)
+	}
+
+	return resMap, nil
+}
+
+// lookupResultsForVoteOptions looks in the CastVote to see how many votes
+// each option has received.
+func (d *decred) lookupResultsForVoteOptions(options []VoteOption) ([]decredplugin.VoteOptionResult, error) {
+
+	results := make([]decredplugin.VoteOptionResult, 0, len(options))
+
+	for _, v := range options {
+		var votes uint64
+		tokenVoteBit := v.Token + strconv.FormatUint(v.Bits, 16)
+		err := d.recordsdb.
+			Model(&CastVote{}).
+			Where("token_vote_bit = ?", tokenVoteBit).
+			Count(&votes).
+			Error
+		if err != nil {
+			return nil, err
+		}
+
+		results = append(results,
+			decredplugin.VoteOptionResult{
+				ID:          v.ID,
+				Description: v.Description,
+				Bits:        v.Bits,
+				Votes:       votes,
+			})
+	}
+
+	return results, nil
+}
+
+// manuallyLookupNewVoteResults manually looks up vote results for records
+// for which voting has started but there is no vote option result in resMap.
+// This is meant to be called after resMap has been populated with the records
+// for which the vote results have been lazily loaded into the VoteResults
+// table.
+func (d *decred) manuallyLookupNewVoteResults(resMap map[string][]decredplugin.VoteOptionResult, svMap map[string]StartVote) error {
+
+	for token, sv := range svMap {
+		_, ok := resMap[token]
+		if ok {
+			continue
+		}
+
+		results, err := d.lookupResultsForVoteOptions(sv.Options)
+		if err != nil {
+			return err
+		}
+
+		resMap[token] = results
+	}
+
+	return nil
+}
+
+func (d *decred) cmdBatchVoteSummary(payload string) (string, error) {
+	log.Tracef("cmdBatchVoteSummary")
+
+	bvs, err := decredplugin.DecodeBatchVoteSummary([]byte(payload))
+	if err != nil {
+		return "", err
+	}
+
+	// This query gets the latest version of each record
+	query := `SELECT a.* FROM records a
+	LEFT OUTER JOIN records b
+		ON a.token = b.token AND a.version < b.version
+	WHERE b.token IS NULL AND a.token IN (?)`
+
+	rows, err := d.recordsdb.Raw(query, bvs.Tokens).Rows()
+
+	if err != nil {
+		return "", err
+	}
+	defer rows.Close()
+
+	recordsMap := make(map[string]Record)
```

Comment:
Style nit.  It's idiomatic to not name variables after their type.

https://dave.cheney.net/2019/01/29/you-shouldnt-name-your-variables-after-their-types-for-the-same-reason-you-wouldnt-name-your-pets-dog-or-cat